backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/StubRenderer.kt

```diff
@@ -18,6 +24,8 @@ object StubRenderer {
                 }
                 +"*/"
             }
+            this.descriptor?.extractKDocString()?.let { +it }
```

Comment:
So, AFAIU, this just copies Kotlin comment to Obj-C headers. I guess this is ok in most cases, but
1. Are there simple cases when this doesn't work (i.e. because of different comment syntax in Kotlin and Obj-C)?
2. How hard would it be to extract "the contents" of Kotlin comment and "wrap" it properly as Obj-C comment?

The latter should also help merging this comment with one emitted above, which somewhat makes sense.

Reply:
1. KDoc is a legal comment in ObjC, I believe. So it  shouldn't cause any compilation issues. However KDoc-specific syntax is not the exact equivalent of HeaderDoc syntax used by Apple. OTOH Doxygen seems to be compatibe with our format.  So, the comment will be legal, readable and partially supported by the common documentation tools.

2. Full syntax translation between KDoc and other tools is hardly possible and not a goal probably. As for simple formatting like new lines, spaces, etc it may be not so trivial too. Particularly leading spaces and ` *. ` at each line not always easy to interpret correctly and/or replace or remove for correct representation. Consider a simple block 
```
/********************************
 * A legal and fancy KDoc here  *
 ********************************/     
```
that may easily become misformatted.

I consider any non-trivial re-formatting as dangerous. At a minimal risk and efforts we can replace KDoc-specific leading `/**` with HeaderDoc compatible `/*!` which is also Doxygen-compatible. However, in that way we'll pretend to provide HeaderDoc syntax inside of it, which may be too brave assumption. Does it make sense? Note that Doxygen supports both formats anyway.

Reply:
Thank you for the detailed examination!
I guess placing the kdoc as is should be ok for now, especially considering the complexity brought by any potential modifications.